include/xtensor-sparse/xlil.hpp

```diff
@@ -52,36 +70,98 @@ namespace xt
         template<class... Args>
         reference access_impl(Args... args);
 
-        template<class... Args>
-        std::pair<list_index_type, index_type> build_keys(Args... args) const;
+        template <class S = shape_type>
+        void reshape_impl(S&& shape, std::false_type);
 
         shape_type m_shape;
+        shape_type m_strides;
+        shape_type m_backstrides;
         container_type m_data;
     };
 
     template<class T>
     const typename xlil_container<T>::value_type xlil_container<T>::ZERO = 0;
 
     template<class T>
-    xlil_container<T>::xlil_container(const shape_type& shape)
+    inline xlil_container<T>::xlil_container(const shape_type& shape)
      : m_shape{shape}
     {}
 
+    template <class T>
+    inline auto xlil_container<T>::size() const noexcept -> size_type
+    {
+        return compute_size(shape());
+    }
+
     template<class T>
-    auto xlil_container<T>::shape() const -> const shape_type&
+    inline auto xlil_container<T>::shape() const -> const shape_type&
     {
         return m_shape;
     }
 
     template<class T>
-    auto xlil_container<T>::dimension() const -> size_type
+    inline auto xlil_container<T>::dimension() const -> size_type
     {
         return m_shape.size();
     }
 
+    template <class T>
+    template <class S>
+    inline void xlil_container<T>::resize(S&& shape, bool force)
+    {
+        std::size_t dim = shape.size();
+        if (m_shape.size() != dim || !std::equal(std::begin(shape), std::end(shape), std::begin(m_shape)) || force)
+        {
+            m_shape = xtl::forward_sequence<shape_type, S>(shape);
+            resize_container(m_strides, dim);
+            resize_container(m_backstrides, dim);
+            compute_strides<layout_type::row_major>(m_shape, XTENSOR_DEFAULT_LAYOUT, m_strides, m_backstrides);
```

Comment:
This should be replaced with an implementation specific to sparse containers.

Reply:
you're right.